peer.go

```diff
@@ -216,9 +216,6 @@ func (l *PeerList) choosePeer(prevSelected map[string]struct{}, avoidHost bool)
 
 // GetOrAdd returns a peer for the given hostPort, creating one if it doesn't yet exist.
 func (l *PeerList) GetOrAdd(hostPort string) *Peer {
-	if ps, ok := l.exists(hostPort); ok {
```

Comment:
it's a bit strange how we ended up doing it twice. should we add a TODO to replace all `GetOrAdd` calls?

Reply:
yeah, same - I feel like it probably happened organically through gradual refactors. will add a TODO